docs/advanced/atomic-swap.md

```diff
@@ -0,0 +1,33 @@
+# Atomic Swaps
+
+---
+
+## Overview
+
+An Atomic Swap is a smart contract technology which makes possible to exchange coins from two different blockchains without having to trust any third party, for example a centralized exchange.
+The smart contracts used in this case are not necessarily Turing-complete smart contracts (like in the system of Ethereum).
+Smart contracts also exist in the system of Bitcoin or Decred, or almost any other cryptocurrency.
+The scripting languages of these cryptocurrencies are not Turing-complete, which is intentionally designed like this to avoid loops, which would create security vulnerabilities.  
+For an atomic swap, a cryptocurrency must have the 'OP_CHECKLOCKTIMEVERIFY' opcode, which makes a timelock possible.
+The other prerequisite is that they need to have the same hash function.
+
+---
+
+## History of Atomic Swaps
+
+Tier Nolan is considered the inventor of atomic swaps. The concept of decentralized exchanges were widely discussed already in 2012. [The first atomic swap](https://blog.decred.org/2017/09/20/On-Chain-Atomic-Swaps/) happened between Decred and Litecoin on September 20, 2017.
```

Comment:
Just to be clear, the LTC/DCR atomic swap described in this blog post was not the first atomic swap in history - it was however the first publicised atomic swap between those two chains.